File: 250212/Run Length 인코딩/run-length-encoding.js
const fs = require("fs");

const string = fs.readFileSync(0).toString().trim();

let idx = 0;
const n = string.length;
let result = "";

while (idx < n) {
    let char = string[idx];
    let count = 0;
    while (idx < n && char == string[idx]) {
        idx++;
        count++;
    }
    result = result + char + count;
}

console.log(result.length);
console.log(result);

File: 250212/Run Length 인코딩/run-length-encoding.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    string str;
    cin >> str;

    string result;
    int idx = 0;

    while (idx < str.length()) {
        char current = str[idx];
        int count = 0;

        while (str[idx] == current) {
            idx++;
            count++;
        }

        result += current;
        result += to_string(count);
    }

    cout << result.length() << "\n" << result << "\n";
    return 0;
}
